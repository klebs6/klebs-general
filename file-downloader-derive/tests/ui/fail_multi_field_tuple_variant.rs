
// file-downloader-derive/tests/ui/fail_multi_field_tuple_variant.rs
use file_downloader::FileDownloader;
use file_downloader_derive::FileDownloader;

#[derive(FileDownloader)]
enum MyEnum {
    // Tuple variant with 2 fields -> should fail unless there's #[download_link = ""] to short-circuit
    TwoField(u32, u32),
}

fn main() {}
