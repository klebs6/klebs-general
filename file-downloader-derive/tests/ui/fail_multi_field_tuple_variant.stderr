error: Multiple fields in tuple variant not supported without a download_link
 --> tests/ui/fail_multi_field_tuple_variant.rs:9:5
  |
9 |     TwoField(u32, u32),
  |     ^^^^^^^^^^^^^^^^^^

error: `match` arm with no body
 --> tests/ui/fail_multi_field_tuple_variant.rs:9:5
  |
9 |     TwoField(u32, u32),
  |     ^^^^^^^^^^^^^^^^^^- help: add a body after the pattern: `=> todo!(),`
